_episodes/09-practical-examples.md

```diff
@@ -91,5 +91,28 @@ keypoints:
 > > ~~~
 > > {: .output}
 > > Congrats! Now we find all the people who are under the drinking age!
+> > 
+> > Now we can make the whole thing even better, so far we only used containers to run our scripts in, but we can also create new containers our colleagues can use. It is good practice to create a folder for each dockercontainer. Copy the respective scripts to the folders
+> > ~~~
+> > $ mkdir convertbirthdaytoage
+> > $ cp convertbirthdaytoage.py convertbirthdaytoage
+> > $ mkdir checkdrinkinglimit
+> > $ cp checkdrinkinglimit.py checkdrinkinglimit
+> > ~~~
+> > {: .language-bash}
+> > The Dockerfiles for the respective folders are then
+> > ~~~
+> > FROM python:2.7
+> > COPY convertbirthdaytoage.py /convertbirthdaytoage.py
+> > ENTRYPOINT ["python2","/convertbirthdaytoage.py"]
+> > ~~~
+> > ~~~
+> > FROM python:3.6
+> > COPY checkdrinkinglimit.py /checkdrinkinglimit.py
+> > ENTRYPOINT ["python3","/checkdrinkinglimit.py"]
+> > ~~~
+> > The `ENTRYPOINT` instruction allows you to configure a container that will run as an executable. This allows arguments to be passed to the entry point, i.e., `docker run <image> -d` will pass the `-d` argument to the entry point. You can override the `ENTRYPOINT` instruction using the `docker run --entrypoint` flag.
```

Comment:
Another solution would be to use CMD.
I wonder if we should explain the difference.

Reply:
I changed my mind. I think you should use `CMD` because `ENTRYPOINT` isn't taught in this course.

Reply:
I thought `CMD` is executed when you build the container?

Reply:
Sorry that was incorrect, if we use `CMD` then I have to specify the command when I run the container, but I only want to specify the parameters.

Reply:
`CMD` is not run when you build the container. Maybe you're thinking of `RUN`.
`CMD` will provide the default command. If you do not specify another command when running the container this one will be run.

I agree that there is a good point to be made to use ENTRYPOINT because it doesn't make sense to run any other command within this container. Entrypoint is not usually overridden so it makes sense to use it.
My reasoning to prefer `CMD` over `ENTRYPOINT` is that `ENTRYPOINT` isn't explained in the course and it might be a little confusing to explain why they should use entrypoint. As long as you don't override the command the effect will be the same.

Reply:
but if I run the container with additional arguments, e.g. the filenames then `CMD` will  still work?

Reply:
ohhh now I get it. No then it wouldn't work with `CMD`. Please keep `ENTRYPOINT` :sweat_smile: